### PR TITLE
feat(web-ui): use Vite instead of react-scripts, update node version

### PR DESCRIPTION
**Pull request by sparrowhe**
**BREAKING CHANGE**

Please upgrade your node to LTS, it's 22.11 for now.

[Vite](https://vitejs.dev) is the new build tool for this project now, it is more fast and can improve development efficiency.

### CODE REVIEW

File: kagami/webui/src/page/Ad/Announcement/styles.module.scss [hunk 1]
```diff
@@ -35,7 +35,7 @@
               // 背景颜色
     box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1); // 阴影效果
     position:relative;
-    background-image: url('../../picture/kagami.jpg'); /* 背景图片 */
+    background-image: url('../../../picture/kagami.jpg'); /* 背景图片 */
```
Comment (wxiwnd) [inline kagami/webui/src/page/Ad/Announcement/styles.module.scss]: I remember "kagami.jpg" is actually a png file. You can rename all the jpg files into png.

File: kagami/webui/src/page/IntroductionPage/FirstSection/index.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import {Link, useNavigate} from 'react-router-dom';
-import LogoImage from '../../picture/logo.jpg';
-import WordImage from '../../picture/word.jpg';
+import LogoImage from '../../../picture/logo.jpg';
+import WordImage from '../../../picture/word.jpg';
```
Comment (wxiwnd) [inline kagami/webui/src/page/IntroductionPage/FirstSection/index.tsx]: btw, use assets instead of picture as folder name?

**Comment by wxiwnd** [comment]
Please solve the conflict and rebase this branch

File: kagami/webui/src/page/IntroductionPage/FirstSection/index.tsx (hunk 2 above)
Comment (sparrowhe) [reply kagami/webui/src/page/IntroductionPage/FirstSection/index.tsx]: done

**Comment by wxiwnd** [comment]
LGTM, @ianzhaoyh please review this pr

**Comment by ianzhaoyh** [comment]
LGTM

**Pull request merged**